### PR TITLE
Sync. BUG fix and witness reward remainder distribution

### PR DESCRIPTION
**Pull request by Andrei-Masilevich**

### CODE REVIEW

File: libraries/chain/database/block_tasks/process_witness_reward_in_sp_migration.cpp [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+        return;
+    }
+
+    blocks_story_service_i& blocks_story_service = ctx.services().blocks_story_service();
+
+    const auto& dpo = dprops_service.get();
+
+    using recipients_type = std::map<account_name_type, share_type>;
+    uint32_t block_num = dpo.head_block_number;
+    optional<signed_block> b;
```
Comment (Katkevich) [inline libraries/chain/database/block_tasks/process_witness_reward_in_sp_migration.cpp]: Put definition near to initialization.

File: libraries/chain/database/block_tasks/process_witness_reward_in_sp_migration.cpp [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+    while (block_num > old_reward_alg_switch_reward_block_num)
+    {
+        b = blocks_story_service.fetch_block_by_number(block_num--);
+        if (b.valid())
+        {
+            signed_block block = *b;
+
+            share_type payment;
+
+            if (total_payment >= migrate_deferred_payment)
```
Comment (Katkevich) [inline libraries/chain/database/block_tasks/process_witness_reward_in_sp_migration.cpp]: Replace this is-else statement with
`share_type payment = std::min(migrate_deferred_payment, total_payment);`
`total_payment -= payment;`

File: libraries/chain/database/block_tasks/process_witness_reward_in_sp_migration.cpp [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+            }
+            else
+            {
+                payment = std::max(total_payment, share_type());
+                total_payment = 0;
+            }
+
+            witnesses[block.witness] += payment;
+
+            if (total_payment == share_type())
```
Comment (Katkevich) [inline libraries/chain/database/block_tasks/process_witness_reward_in_sp_migration.cpp]: Put this break-condition into 'while' condition

File: libraries/chain/database/block_tasks/process_witness_reward_in_sp_migration.cpp [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+        = ctx.services().witness_reward_in_sp_migration_service();
+
+    if (!witness_reward_in_sp_migration_service.is_exists())
+    {
+        return;
+    }
+
+    blocks_story_service_i& blocks_story_service = ctx.services().blocks_story_service();
+
+    const auto& dpo = dprops_service.get();
```
Comment (alexplevako) [inline libraries/chain/database/block_tasks/process_witness_reward_in_sp_migration.cpp]: rename pleas `dpo` to `dprops`

File: libraries/chain/database/block_tasks/process_witness_reward_in_sp_migration.cpp [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+
+    using recipients_type = std::map<account_name_type, share_type>;
+    uint32_t block_num = dpo.head_block_number;
```
Comment (alexplevako) [inline libraries/chain/database/block_tasks/process_witness_reward_in_sp_migration.cpp]: `uint32_t block_num =  dprops_service.get().head_block_number;`

File: libraries/chain/database/block_tasks/process_witness_reward_in_sp_migration.cpp [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+    {
+        wlog("Migration rest is not enough. ${n} blocks left",
+             ("n", block_num - old_reward_alg_switch_reward_block_num));
+    }
+    if (total_payment > 0)
+    {
+        wlog("Migration rest is too large. Left ${r}", ("r", asset(total_payment, SP_SYMBOL)));
+    }
+
+    for (const auto& witness_p : witnesses)
```
Comment (alexplevako) [inline libraries/chain/database/block_tasks/process_witness_reward_in_sp_migration.cpp]: bad name for variable `witness_p`

**Pull request merged**